src/api/apiClient.ts

```diff
@@ -35,5 +36,61 @@ axiosClient.interceptors.response.use(
     },
 )
 
+export const errorMap = (issue: ZodIssueOptionalMessage, ctx: ErrorMapCtx) => {
+    if (!ctx.data) {
+        return {
+            message: `${issue.path.join('.')} is a required field`,
+        }
+    }
+
+    switch (issue.code) {
+        case ZodIssueCode.too_small:
+            if (issue.type === 'string') {
+                return {
+                    message:
+                    `Invalid value: ${issue.path.join('.')} length must be >= ${issue.minimum} but got ${ctx.data}`,
+                }
+            } else {
+                return {
+                    message: 
+                    `Invalid value: ${issue.path.join('.')} must be >= ${issue.minimum} but got ${ctx.data}`,
+                }
+            }
+        case ZodIssueCode.too_big:
+            if (issue.type === 'string') {
+                return {
+                    message:
+                    `Invalid value: ${issue.path.join('.')} length must be <= ${issue.maximum} but got ${ctx.data}`,
+                }
+            } else {
+                return {
+                    message: 
+                    `Invalid value: ${issue.path.join('.')} must be <= ${issue.maximum} but got ${ctx.data}`,
+                }
+            }
+        case ZodIssueCode.invalid_string:
+            if (issue.validation === 'regex') {
+                return {
+                    message: `Invalid value: format of ${issue.path.join('.')} is invalid.`,
```

Comment:
another nitpick, but considering all of these return an object with a message property prefixed by `Invalid value:` we could make a tiny util function within the errorMap to abstract that
```
(message: string) => ({ message: `Invalid value: ${message}` })
```

Reply:
i just changed it to only print the 'Invalid value' if we don't know the details of the error and we're just blindly print the message, like:
`Invalid value: ${issue.path.join('.')}. ${issue.message || ''}`

but for something like `key length must be greater than X`, i'm thinking the 'invalid value' is a bit redundant?